### PR TITLE
AP_RangeFinder: save little flash by using const string

### PR DESCRIPTION
**Pull request by ak-ironix**
Hey there! I am a beginner at ArduPilot. I just found this code where _Rangefinder_ was being used multiple times in `sprintf` calls. We can save some flash by storing it in a const char array and then using it wherever needed.
Can anyone please review it?
Thanks in advance! Looking forward to make more contributions to ArduPilot.

### CODE REVIEW

File: libraries/AP_RangeFinder/AP_RangeFinder.cpp [hunk 1]
```diff
@@ -838,9 +838,11 @@ bool RangeFinder::prearm_healthy(char *failure_msg, const uint8_t failure_msg_le
         if ((Type)params[i].type.get() == Type::NONE) {
             continue;
         }
+        
+        const char str[] = "Rangefindr ";
 
         if (drivers[i] == nullptr) {
-            hal.util->snprintf(failure_msg, failure_msg_len, "Rangefinder %X: Not Detected", i + 1);
+            hal.util->snprintf(failure_msg, failure_msg_len, "%s %X: Not Detected", str, i + 1);
```
Comment (IamPete1) [inline libraries/AP_RangeFinder/AP_RangeFinder.cpp:845]:
```suggestion
            hal.util->snprintf(failure_msg, failure_msg_len, "%s%X: Not Detected", "Rangefinder ", i + 1);
```

File: libraries/AP_RangeFinder/AP_RangeFinder.cpp [hunk 2]
```diff
@@ -872,10 +874,10 @@ bool RangeFinder::prearm_healthy(char *failure_msg, const uint8_t failure_msg_le
 
         switch (drivers[i]->status()) {
         case Status::NoData:
-            hal.util->snprintf(failure_msg, failure_msg_len, "Rangefinder %X: No Data", i + 1);
+            hal.util->snprintf(failure_msg, failure_msg_len, "%s %X: No Data", str, i + 1);
```
Comment (IamPete1) [inline libraries/AP_RangeFinder/AP_RangeFinder.cpp:877]:
```suggestion
            hal.util->snprintf(failure_msg, failure_msg_len, "%s%X: No Data", "Rangefinder ", i + 1);
```

File: libraries/AP_RangeFinder/AP_RangeFinder.cpp [hunk 3]
```diff
@@ -872,10 +874,10 @@ bool RangeFinder::prearm_healthy(char *failure_msg, const uint8_t failure_msg_le
...
             return false;
         case Status::NotConnected:
-            hal.util->snprintf(failure_msg, failure_msg_len, "Rangefinder %X: Not Connected", i + 1);
+            hal.util->snprintf(failure_msg, failure_msg_len, "%s %X: Not Connected", str, i + 1);
```
Comment (IamPete1) [inline libraries/AP_RangeFinder/AP_RangeFinder.cpp:880]:
```suggestion
            hal.util->snprintf(failure_msg, failure_msg_len, "%s%X: Not Connected", "Rangefinder ", i + 1);
```

**Review by IamPete1 — COMMENTED** [review]
No need to actually make a separate variable, the compiler is clever enough to combine common strings.

File: libraries/AP_RangeFinder/AP_RangeFinder.cpp (hunk 1 above)
Comment (ak-ironix) [reply libraries/AP_RangeFinder/AP_RangeFinder.cpp]: Okay. Thanks for suggesting the change.

**Comment by ak-ironix** [comment]
@IamPete1 Thanks for suggesting the changes. Please check if its mergeable now.

**Comment by IamPete1** [comment]
It doesn't actually save flash any more.. Maybe the compiler is not as smart as I thought....

**Comment by ak-ironix** [comment]
No problem. I have reverted the changes. Please check if they are good.

**Comment by IamPete1** [comment]
Now it costs 8 bytes, that missing letter in rangfinder must have been what was causing the saving in the original PR. Compilers are mysterious sometimes, maybe tying to tell us its not worth it ;).

BTW you can see the cost in the `test size` CI job. https://github.com/ArduPilot/ardupilot/actions/runs/3257277467/jobs/5348444180

**Comment by ak-ironix** [comment]
Alright, thanks for letting me know. I'll close the PR.

**Pull request closed without merging**